test/test_main.py

```diff
@@ -130,34 +158,53 @@ def test_event_reemitted(self):
         }))
 
         # First run "install" to make sure all hooks are set up.
-        state = self._simulate_event('install', charm_config)
+        state = self._simulate_event(SimulatedEventSpec(InstallEvent, 'install'), charm_config)
         self.assertEqual(state['observed_event_types'], [InstallEvent])
 
-        state = self._simulate_event('config-changed', charm_config)
+        state = self._simulate_event(SimulatedEventSpec(ConfigChangedEvent, 'config-changed'), charm_config)
         self.assertEqual(state['observed_event_types'], [ConfigChangedEvent])
 
         # Re-emit should pick the deferred config-changed.
-        state = self._simulate_event('update-status', charm_config)
+        state = self._simulate_event(SimulatedEventSpec(UpdateStatusEvent, 'update-status'), charm_config)
         self.assertEqual(state['observed_event_types'], [ConfigChangedEvent, UpdateStatusEvent])
 
     def test_multiple_events_handled(self):
-
         # Sample events with a different amount of dashes used
         # and with endpoints from different sections of metadata.yaml
-        events_under_test = {
-            'install': InstallEvent,
-            'update_status': UpdateStatusEvent,
-            'leader_settings_changed': LeaderSettingsChangedEvent,
-            'db_relation_joined': RelationJoinedEvent,
-            'mon_relation_changed': RelationChangedEvent,
-            'ha_relation_broken': RelationBrokenEvent,
-        }
-
-        expected_event_data = {
-            'db_relation_joined': {'relation_name': 'db', 'relation_id': 1, 'unit_name': 'remote/0'},
-            'mon_relation_changed': {'relation_name': 'mon', 'relation_id': 2, 'unit_name': 'remote/0'},
-            'ha_relation_broken': {'relation_name': 'ha', 'relation_id': 3},
-        }
+        events_under_test = (
```

Comment:
This should be a list instead of a tuple.

Also, this layout tends to create better looking and more compact code:
```
some_list = [(
        foo,
        bar,
), (
        ... etc ...
)]
```